### PR TITLE
Add New Relic via the Gorelic platform agent

### PR DESCRIPTION
**Pull request by relistan**
This adds basic monitoring of the Http transactions, the key functions of the service, and the Redis protocol server endpoints that matter. Currently reporting using the Gorelic platform agent since there are ongoing licensing issues with the New Relic Go agent.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -55,6 +55,18 @@ AWS environment variables:
  * `AWS_ACCESS_KEY_ID`
  * `AWS_SECRET_ACCESS_KEY`
 
+If you are a New Relic customer and wish to monitor this using New Relic's
+service, the service includes the
+[Gorelic](https://github.com/yvasiyarov/gorelic) platform agent.  Thi is
```
Comment (mihaitodor) [inline README.md]: Typo.

File: main.go [hunk 2]
```diff
@@ -111,6 +112,30 @@ func handleSignals(fCache *filecache.FileCache, mList *memberlist.Memberlist) {
...
 
+// configureNewRelic sets up and starts a Gorelic agent if we have a
+// New Relic license available.
+func configureNewRelic() *gorelic.Agent {
+	nrLicense := os.Getenv("NEW_RELIC_LICENSE_KEY")
+	var agent *gorelic.Agent
+	if nrLicense == "" {
+		log.Info("No New Relic license found, not starting an agent")
+
+		return nil
```
Comment (mihaitodor) [inline main.go]: Won't the accesses to `agent.Tracer` freak out if the agent is nil?

File: main.go [hunk 3]
```diff
@@ -111,6 +112,30 @@ func handleSignals(fCache *filecache.FileCache, mList *memberlist.Memberlist) {
...
+	}
+
+	agent = gorelic.NewAgent()
+	svcName := os.Getenv("SERVICE_NAME")
+	envName := os.Getenv("ENVIRONMENT_NAME")
+	if svcName != "" && envName != "" {
+		agent.NewrelicName = fmt.Sprintf("%s-%s", svcName, envName)
+	}
+	agent.Verbose = true
```
Comment (mihaitodor) [inline main.go]: Stuff might get noisy in prod... Maybe have this enabled through an env var.

File: main.go (hunk 2 above)
Comment (relistan) [reply main.go]: I worked around that in the Http code, but looks like the Redis code doesn't work well

File: main.go (hunk 3 above)
Comment (relistan) [reply main.go]: I want to see what it does in dev, and will turn it off if it's noisy. We don't need a setting for that as I can't see why we'd need to flip it on/off  99% of the time.

**Pull request merged**